Clean out Ubuntu 16.04 testing

dougbu opened the pull request:
- use Ubuntu 18.04 agents for testing in Docker containers on Helix

nit: clean up a few old mentions of Centos and Fedora testing

dougbu committed:
Clean out Ubuntu 16.04 testing
- use Ubuntu 18.04 agents for testing in Docker containers on Helix

nit: clean up a few old mentions of Centos and Fedora testing

dougbu commented on eng/targets/Helix.targets:
```diff
@@ -9,7 +9,18 @@
+      Windows.81.Amd64.Open;
+      Redhat.7.Amd64.Open;
+      Redhat.7.Amd64;
+      Debian.9.Amd64.Open;
+      Debian.9.Amd64;
```
`Debian.9.Amd64.Open` was previously duplicated.

dougbu commented on eng/targets/Helix.targets:
```diff
@@ -9,7 +9,18 @@
+      Ubuntu.2004.Amd64;
+      (Alpine.312.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.12-helix-20200908125345-56c6673;
+      (Fedora.33.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-33-helix-20210120000908-a9df267
+    </SkipHelixQueues>
```
@HaoK why are some internal queues listed and others are not❔ The full Helix matrix uses `'$(_UseHelixOpenQueues)' == 'true'` checks to avoid using any of the listed internal queues. But, `Windows.7.Amd64` and `Windows.81.Amd64` queues exist, as does `Ubuntu.1804.Amd64` for the Docker containers.

Is the mix historical❔

HaoK replied:
We don't run helix on internal queues at all anymore do we?  I'd guess this is just legacy

dougbu commented on eng/scripts/RunHelix.ps1:
```diff
@@ -8,13 +8,12 @@
     Ubuntu.1804.Amd64.Open
-    Windows.10.Amd64.Open
+    Ubuntu.2004.Amd64.Open
+    Windows.10.Amd64.Server20H2.Open
```
`Windows.10.Amd64.Open` exists but we use `Windows.10.Amd64.Server20H2.Open` by default in real life. Matching that here.

dougbu committed:
nits:
- fix `%(Platform)` metadata for OSX `@(HelixAvailableTargetQueue)` items
- don't mention unused internal queues in `$(TestDependsOnPlaywright)` update of `$(SkipHelixQueues)`
- sort remaining queues alphabetically in `$(SkipHelixQueues)`

dougbu commented on eng/targets/Helix.Common.props:
```diff
@@ -19,13 +19,12 @@
   <ItemGroup Condition="'$(TargetArchitecture)' == 'x64' AND '$(IsHelixDaily)' == 'true' AND '$(_UseHelixOpenQueues)' == 'true' AND '$(IsWindowsOnlyTest)' != 'true'">
-    <HelixAvailableTargetQueue Include="Ubuntu.1604.Amd64.Open" Platform="Linux" />
     <HelixAvailableTargetQueue Include="Ubuntu.2004.Amd64.Open" Platform="Linux" />
     <HelixAvailableTargetQueue Include="OSX.1100.Amd64.Open" Platform="Linux" />
```
Why are the two `OSX.*` queues using `Platform="Linux"`❔ We have an `OSX` `@(HelixAvailablePlatform)` item and Libuv.BindTests.csproj attempts to exclude it.

HaoK replied:
Oversight most likely, when things started we mostly just had windows and non windows queues.  There wasn't anything that used to care about linux vs osx

dougbu commented:
The https://dev.azure.com/dnceng/public/_build/results?buildId=1150252&view=results run passed in the job this PR affects. Ready for review&hellip;

dougbu commented:
/backport to release/3.1

dougbu commented:
/backport to release/5.0
